src/Components/WebAssembly/WebAssembly.Authentication/src/Services/DeserializedAuthenticationStateProvider.cs

```diff
@@ -28,12 +31,18 @@ internal sealed class DeserializedAuthenticationStateProvider : AuthenticationSt
     [DynamicDependency(JsonSerialized, typeof(ClaimData))]
     public DeserializedAuthenticationStateProvider(PersistentComponentState state, IOptions<AuthenticationStateDeserializationOptions> options)
     {
-        if (!state.TryTakeFromJson<AuthenticationStateData?>(PersistenceKey, out var authenticationStateData) || authenticationStateData is null)
+        // For backwards compatibility, try to read from the direct state first
+        if (AuthStateData is null && 
```

Comment:
`TryTakeFromJson` is performed by persistence mechanism that is triggered by adding `SupplyParameterFromPersistentComponentState` attribute to a property. We don't have to do it explicitly. Required changes that we should apply to `DeserializedAuthenticationStateProvider` and `AuthenticationStateSerializer` should resemble these done in https://github.com/dotnet/aspnetcore/pull/60634 to src/Components/Endpoints/src/Forms/EndpointAntiforgeryStateProvider.cs and src/Components/Shared/src/DefaultAntiforgeryStateProvider.cs.